Tests/LLBBuildSystemUtilTests/LocalExecutorTests.swift

```diff
@@ -0,0 +1,137 @@
+                guard case let LLBLocalExecutorError.unexpected(underlyingError) = error else {
+                    XCTFail("Unexpected error type")
+                    return
+                }
+
+                XCTAssertEqual(String(describing: underlyingError), "could not find executable for 'racism'")
+            }
+        }
+    }
+
+    func testActionFailure() throws {
+        try withTemporaryDirectory { tempDirectory in
+            let localExecutor = LLBLocalExecutor(outputBase: tempDirectory)
+            let testEngineContext = LLBTestBuildEngineContext()
+
+            let request = LLBActionExecutionRequest.with {
+                $0.actionSpec = .with {
+                    $0.arguments = ["/bin/bash", "-c", "false"]
+                }
+            }
+
+            let result = try localExecutor.execute(request: request, engineContext: testEngineContext).wait()
+            XCTAssertEqual(result.exitCode, 1)
+            XCTAssert(result.outputs.isEmpty)
+        }
+    }
+
+    func testMissingOutputFile() throws {
+        try withTemporaryDirectory { tempDirectory in
+            let localExecutor = LLBLocalExecutor(outputBase: tempDirectory)
+            let testEngineContext = LLBTestBuildEngineContext()
+
+            let request = LLBActionExecutionRequest.with {
+                $0.actionSpec = .with {
+                    $0.arguments = ["/bin/bash", "-c", "true"]
+                }
+                $0.outputs = [
+                    .with {
+                        $0.path = "some/path"
+                        $0.type = .file
+                    }
+                ]
+            }
+
+            XCTAssertThrowsError(try localExecutor.execute(request: request, engineContext: testEngineContext).wait()) { error in
+                guard case let LLBLocalExecutorError.unexpected(underlyingError) = error,
+                      let fsError = underlyingError as? FileSystemError else {
+                    XCTFail("Unexpected error type")
+                    return
+                }
+
+                XCTAssertEqual(fsError, FileSystemError.noEntry)
+            }
+        }
+    }
+
+    func testMissingOutputDirectory() throws {
+        try withTemporaryDirectory { tempDirectory in
+            let localExecutor = LLBLocalExecutor(outputBase: tempDirectory)
+            let testEngineContext = LLBTestBuildEngineContext()
+
+            let request = LLBActionExecutionRequest.with {
+                $0.actionSpec = .with {
+                    $0.arguments = ["/bin/bash", "-c", "true"]
+                }
+                $0.outputs = [
+                    .with {
+                        $0.path = "some/path"
+                        $0.type = .directory
+                    }
+                ]
+            }
+
+            let result = try localExecutor.execute(request: request, engineContext: testEngineContext).wait()
```

Comment:
@dmbryson this test takes more than a couple seconds, it is enforcing that a missing directory being imported creates an empty tree, but the LLBCASFileTree.import method takes a long time to error out. Do you know what could be?

Reply:
🤔 not sure, I’ll take a look